src/web/app/components/question-types/question-edit-answer-form/constsum-recipients-question-edit-answer-form.component.ts

```diff
@@ -28,8 +28,8 @@ export class ConstsumRecipientsQuestionEditAnswerFormComponent
    */
   triggerResponse(event: number): void {
     if (!event) {
```

Comment:
I think the problem here is the check (not the empty array thing). The check says that if user want to delete the response, they can use backspace to delete the answer (though it is not possible to pass the constraint check). If `event` is `0` which is valid answer, it will incorrectly enter this branch as `!0` is `true` in JavaScript.

Reply:
> The check says that if user want to delete the response, they can use backspace to delete the answer

This should not be possible/allowed because the response is part of a larger question set. Deleting an option (currently represented as a response) while still allowing other options to be present should not logically be possible as they collectively constitute an answer set. This is my opinion anyway.

If I'm not wrong, since this method is only triggered `onChange`, it will not set the `answer` to `0` unless a change was made by the user. This should be a sufficient enough condition to indicate that the user wanted to give the 0 response. Additionally, with the placeholder, there is sufficient visual feedback to the user that the response is not 'empty' per se.

Reply:
Then I think the correct fix is to remove the check. I think we should also remove the placeholder (see below).

Here are the correct behaviors in my mind:
- User leaves it as blank -> nothing being stored (which is why the placeholder thing should be removed or will be mis-interpreted)
- User enters zero (will treat it as valid response) (because of the removal of the check)
- User cannot delete submitted response for the question (which follows your reasoning)

Reply:
I see okay, yes that makes more sense thank you for the explanation

Reply:
> * User leaves it as blank -> nothing being stored (which is why the placeholder thing should be removed or will be mis-interpreted)

@xpdavid What happens if a user enters values for some options but not others? I think currently we assign 0 to empty entries because we need to consider _all_ options in the calculation, due to the nature of this question.

Reply:
> What happens if a user enters values for some options but not others?

The validation would fail (empty response will treat as zero when doing validation).

Reply:
> The validation would fail (empty response will treat as zero when doing validation).

I'm assuming you mean backend validation here. The problem is that an empty response will never be sent to the backend because of the frontend logic:

https://github.com/TEAMMATES/teammates/blob/39013a40280b004a6c639efb4c4fa348719655e9/src/web/app/pages-session/session-submission-page/session-submission-page.component.ts#L542-L544

Reply:
I am actually talking about frontend :P
https://github.com/TEAMMATES/teammates/blob/39013a40280b004a6c639efb4c4fa348719655e9/src/web/app/components/question-types/question-constraint/constsum-recipients-question-constraint.component.ts#L63-L71

We do assign 0 to empty entries

Reply:
This is only in the constraint component though. This method is only used by the component's html for user indication of whether the constraints have been met. The constraints themselves are not checked during session submission, thus why we have https://github.com/TEAMMATES/teammates/issues/10397

Reply:
> If this were true, wouldn't this then also naturally extend to the fact that users cannot delete their responses?

Thought we have agreed on this. Yes for this question.

> Assigning zero to empty entries no longer makes it empty?

Just for sake of calculating the total and display the error message. None of this happens in backend and persist to the database.